include/application_defined_privileged_functions.h

```diff
@@ -22,25 +22,75 @@
 #ifndef APPLICATION_DEFINED_PRIVILEGED_FUNCTIONS_H_
 #define APPLICATION_DEFINED_PRIVILEGED_FUNCTIONS_H_
 
-void reboot_system(char type) {
-    prvRaisePrivilege();
+#include "FreeRTOS.h"
+#include "HL_reg_system.h"
+#include "bl_eeprom.h"
+
 
+void reboot_system(char type) {
+    RAISE_PRIVILEGE;
+    eeprom_init();
     switch(type) {
     case 'A':
+        eeprom_set_boot_type('A');
         ex2_log("Rebooting Application");
         break;
     case 'B':
+        eeprom_set_boot_type('B');
         ex2_log("Rebooting to Bootloader");
         break;
     case 'G':
+        eeprom_set_boot_type('G');
         ex2_log("Rebooting to Golden Image");
         break;
     default:
+        eeprom_shutdown();
         ex2_log("Invalid reboot type");
-        break;
+        return;
     }
-
+    eeprom_shutdown();
     systemREG1->SYSECR = (0x10) << 14;
+    RESET_PRIVILEGE;
+}
+
+bool init_eeprom() {
+    RAISE_PRIVILEGE;
+    bool success = eeprom_init();
+    RESET_PRIVILEGE;
+    return success;
+}
+
+void shutdown_eeprom() {
+    RAISE_PRIVILEGE;
+    eeprom_shutdown();
+    RESET_PRIVILEGE;
 }
 
+image_info priv_eeprom_get_app_info() {
+    RAISE_PRIVILEGE;
+    image_info app_info = eeprom_get_app_info();
+    RESET_PRIVILEGE;
+    return app_info;
+}
+
+image_info priv_eeprom_get_golden_info() {
+    RAISE_PRIVILEGE;
+    image_info app_info = eeprom_get_app_info();
+    RESET_PRIVILEGE;
+    return app_info;
+}
```

Comment:
Are these functions supposed to be the same?

Reply:
Good catch. They are not